packages/network-of-terms-catalog/catalog/datasets/gtaa_persoonsnamen.jsonld

```diff
@@ -0,0 +1,37 @@
+{
+  "@context": "https://schema.org/docs/jsonldcontext.jsonld",
+  "@id": "http://data.beeldengeluid.nl/gtaa/Persoonsnamen",
+  "@type": "Dataset",
+  "name": [
+    {
+      "@language": "nl",
+      "@value": "GTAA Persoonsnamen"
+    }
+  ],
+  "creator": [
+    {
+      "@id": "https://www.beeldengeluid.nl/"
+    }
+  ],
+  "url": [
+    "http://data.beeldengeluid.nl/gtaa/Persoonsnamen"

```

Comment:
Is this indeed correct, that is do all Persoonsnamen concepts begin with this URL prefix? If so, this will solve #492.

Reply:
This does not seem to be the case, e.g. https://data.beeldengeluid.nl/gtaa/30369 of Genre Filmmuseum does not start with `http://data.beeldengeluid.nl/gtaa/Genre_Filmmuseum` (the URL that is defined below for this dataset).

Reply:
Concept URI's all start with the same prefix: http://data.beeldengeluid.nl/gtaa/{some_id}
So, Person names don't have a different prefix that geographical names, for example.

Reply:
The dataset’s `schema:url` must be its URL prefix, so that would be `http://data.beeldengeluid.nl/gtaa/` for all datasets, if I understand you correctly. This prefix is used for looking up terms by their URI. Unfortunately, if all datasets share the same prefix, #492 will not be solved. Is there perhaps a generic SPARQL endpoint for lookups in all datasets, or do lookups need to go to the individual endpoints?

(Our property name `schema:url` is slightly unclear, but we decided to stick with pure Schema.org for now, even if it’s less expressive. See also #491.)

Reply:
It doesn't seem right to use the schema:url property for the prefix. The schema:url is the address for the resource. Because the GTAA parts are organised around ConceptSchemes, In our case the ConceptScheme address is the resource. 

But, one could also argue that the Dataset is the resource, and therefore the schema:url should be
``  "url": [
    "https://data.beeldengeluid.nl/id/dataset/0010"
  ],
`` and maybe the ID should also be:
``
  "@id": "https://data.beeldengeluid.nl/id/dataset/0010",
  "@type": "Dataset",
``

I would prefer to use the dataset ID as the schema:``@id``, and the ConceptScheme URI as schema:url

``
  "@id": "https://data.beeldengeluid.nl/id/dataset/0010",
  "@type": "Dataset",
``
``  "url": [
    "http://data.beeldengeluid.nl/gtaa/OnderwerpenBenG"
  ],
`` 

But, then there is no prefix for lookup. Why is that necessary anyway? The Concepts all have a proper URI, so why would you need a prefix? For your information: the GTAA Concepts are all dereferanceable using the URI. Isn't that enough for lookup? Also, no basic auth is required for that.

Reply:
Sorry, my response might have unintentionally sounded a bit agitated at the end. I just wondered why you need the prefix.

Reply:
I’ve just [added some documentation](https://github.com/netwerk-digitaal-erfgoed/network-of-terms/tree/master/packages/network-of-terms-catalog#data-model) to explain the need for the prefix. The Network of Terms has its own lookup functionality to present the same SKOS representation to clients as when they search for terms. Lookup is needed because:

- some sources dereference URIs correctly but return it in their own vocabulary, such as Schema.org, rather than SKOS, which clients expect from the Network of Terms
- other sources use SKOS internally but fail to dereference URIs, including GTAA (as of now):

      $ curl -v -H Accept:application/ld+json https://data.beeldengeluid.nl/gtaa/30369
      < HTTP/1.1 500 Internal Server Error
      
      Could not find acceptable representation

- even if sources dereference correctly and use SKOS, they may return values different from those returned by the Network of Terms search, which can be confusing for clients.

To know which term to source for a certain URI, the Network of Terms needs to know those prefixes.

Does that clear things up?

Reply:
Please note that the GTAA URI's all have the HTTP protocol. We should provide a better return code when using HTTPS for dereferencing.

``To know which term to source for a certain URI, the Network of Terms needs to know those prefixes.``
I still don't get it. Why do you need the prefix? For all sources it is different what the path after that prefix is. In our case the skos:notation value is always corresponding to it: ``/gtaa/{skos:notation}``. But that is not what is needed to dereference the concept. You can use the whole Concept URI (always HTTP) to get the data. Also, there is an API that can be used to get the concept data, but that is not involved at all in the network of terms. Given the queries in ``catalog/queries/lookup/`` and the documentation that you added: ``lookup queries retrieve a single term based on its URI.`` I would deduct that the SPARQL endpoint is used to get the Concept data. Why define a prefix then? It is just our GTAA namespace, not an API function.

Reply:
> Please note that the GTAA URI's all have the HTTP protocol.

Ah, good to know! However, `curl -v -L -H Accept:application/ld+json http://data.beeldengeluid.nl/gtaa/30369` still returns a 500 error.

Anyway, the lookup feature works like this. On https://termennetwerk.netwerkdigitaalerfgoed.nl/lookup, a user enters any term’s URI, for example `http://vocab.getty.edu/aat/300046300` or `http://data.beeldengeluid.nl/gtaa/30369`. 

The question is: how does the Network of Terms know _which SPARQL endpoint to consult_ for each URI? This is what we need the URI prefix for:

- `http://vocab.getty.edu/aat/...` goes to the AAT endpoint;
- `http://data.beeldengeluid.nl/gtaa/...` to the GTAA one.

Does that make sense?

Reply:
>Ah, good to know! However, curl -v -L -H Accept:application/ld+json http://data.beeldengeluid.nl/gtaa/30369 still returns a 500 error.

Sorry to hear that. Now that you bring it up, I realize that I brought up that issue some time ago. It is acknowledged as a bug that it doesn't provide JSON-LD. It will give you the XML representation: ``curl  -L -H Accept:application/rdf+xml http://data.beeldengeluid.nl/gtaa/30369

Reply:
>if the URI begins with http://data.beeldengeluid.nl/gtaa/..., retrieve it from the GTAA endpoint (and so on).
Does that make sense?

I was under the impression that the lookup function was a type of query 'per source'. Now I understand that you do some delegation of the concept URI dereferencing 'by using the schema prefix'. In that case, for any of the GTAA sources it holds that any GTAA URI can be resolved and then ``http://data.beeldengeluid.nl/gtaa`` is the prefix for ALL GTAA sources. And they all use the same SPARQL endpoint, so all concepts are in all endpoints. 
The difficulty for the network of terms is that if many sources share the same prefix, which one does it need to choose?
Luckily, the solution is easy: If we make sure all GTAA lookup queries are not filtering on ConceptScheme it doesn't matter what source you choose: any GTAA source is good.
So, if you agree that this is the solution for the network of terms, than I will make the changes and commit. This means I remove all the triples like this: ``    ?uri skos:inScheme <http://data.beeldengeluid.nl/gtaa/GeografischeNamen> . ``

Reply:
Please go ahead! You can then also reduce the lookup queries to a single one, and use that from all datasets. (I’ve already done this for the search queries.)

Reply:
All GTAA lookup queries share the same file now.